storage/store.go

```diff
@@ -1382,14 +1387,25 @@ func (s *Store) resolveWriteIntentError(ctx context.Context, wiErr *proto.WriteI
 			},
 		}
 		resolveReply := &proto.InternalResolveIntentResponse{}
-		// Add resolve command with wait=false to add to Raft but not wait for completion.
-		waitForResolve := wait && i == len(wiErr.Intents)-1
-		if resolveErr := rng.AddCmd(ctx, proto.Call{Args: resolveArgs, Reply: resolveReply}, waitForResolve); resolveErr != nil {
-			if log.V(1) {
-				log.Warningc(ctx, "resolve for key %s failed: %s", intent.Key, resolveErr)
+
+		go func(errCh chan<- error, key proto.Key) {
+			if resolveErr := rng.AddCmd(ctx, proto.Call{Args: resolveArgs, Reply: resolveReply}, true); resolveErr != nil {
+				if log.V(1) {
+					log.Warningc(ctx, "resolve for key %s failed: %s", key, resolveErr)
+				}
+				errCh <- resolveErr
 			}
-			return resolveErr
-		}
+			wg.Done()
+		}(errChan, intent.Key)
+	}
+
+	go func() {
+		wg.Wait()
```

Comment:
Instead of combining a WaitGroup and a channel, I think it would be simpler to have the AddCmd goroutines write to errCh whether their error is nil or not. Then this goroutine could go away and the final loop could be 

    for i :=0; i < numIntents; i++ { 
        err := <-errChan
        if err != nil {
            return err
        }
    }